packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts

```diff
@@ -152,11 +152,24 @@ export class ApiFetchClient implements ApiClient {
     } catch (e: any) {
       exception = new ResponseJSONParseError(e.message || 'Fail to parse response body', response && response.status || 0, body, {apiName, operationId, url, origin});
     }
+    let statusCodeReviver: ReviverType<T> | undefined = typeof reviver === 'function' || typeof reviver === 'undefined' ?
+      reviver : typeof response?.status !== 'undefined' && reviver[response.status] || undefined;
+    if (!!response?.status && response.status !== 204 && typeof reviver === 'object' && !Object.hasOwn(reviver, response?.status)) {
+      const fallback = Object.entries(reviver)?.reduce((acc: {statusCode: number; reviver: ReviverType<T> | undefined}, [key, value]) => {
+        if (+key < acc.statusCode && +key !== 204) {
```

Comment:
should we check that we fallback to a statusCode in the same category ? (2xx, 3xx, 4xx, ...) ?

Reply:
this is only for the success status code, in the other case, we handle them as error (well not sure for 3xx but we would not revive it anyways)